src/coreclr/jit/codegenarm64.cpp

```diff
@@ -120,7 +120,7 @@ bool CodeGen::genInstrWithConstant(instruction ins,
         // generate two or more instructions
 
         // first we load the immediate into tmpReg
-        instGen_Set_Reg_To_Imm(size, tmpReg, imm);
+        instGen_Set_Reg_To_Imm(EA_8BYTE, tmpReg, imm);
```

Comment:
There is a similar bug in codegenarm.cpp. Do you mind fixing that as well?
For consistency, in both implementation you can use `EA_PTRSIZE`

Reply:
Sure, done.